Automattic/wp-calypso
# Auto-renew: make UI agree with data for auto-renew state (#51996)

Related to #51996 

#### Proposed Changes

1. (fix) Keep the UI and data model in sync for auto-renew state
2. Show a warning when auto-renew is enabled without a payment method

The UI now looks like this:

At list view path `/purchases/subscriptions/DOMAINNAME`

![image](https://user-images.githubusercontent.com/6851384/180178365-51c12cce-b32f-49bc-a921-cdfad50f296f.png)

At detail view path `/purchases/subscriptions/DOMAINNAME/PURCHASEID`
![image](https://user-images.githubusercontent.com/6851384/180178326-63d22256-7cf6-4057-8140-4292207c3a07.png)


![image](https://user-images.githubusercontent.com/6851384/177793349-a61ad252-5b8e-47c1-a038-1bdb36f239b9.png)


#### Testing Instructions

I've tested this on macOS/Chrome using a Pro Plan purchase for my simple blog test site. You can test it using the store sandbox as covered in PCYsg-IA-p2. 

1. Purchase a plan using the test card
2. Check that auto-renew is enabled in both the UI and data model (see linked issue #51996 )
3. Delete payment card
4. Check that auto-renew is still enabled on purchase but you should now see a warning for the payment method as shown above on the list and detail pages
5. Disable auto-renew via toggle and confirm warnings are gone on list and detail pages
6. Add a new payment method with "Use this payment method for all subscriptions on my account" checked. 
7. Enable auto-renew, which should now be allowed, and confirm the card is shown under "payment method"
